packages/next/src/build/entries.ts

```diff
@@ -86,7 +87,10 @@ export function createPagesMapping({
         return result
       }
 
-      const pageKey = getPageFromPath(pagePath, pageExtensions)
+      let pageKey = getPageFromPath(pagePath, pageExtensions)
+      if (isAppRoute) {
+        pageKey = pageKey.replaceAll('%5F', '_')
```

Comment:
replaceAll is availble since node 16, we're still supprting node 14.x, shall we change to replace?